File: 2015/1/puzzle.py
print("Day 1: Not Quite Lisp")

GO_UP = "("

with open('input.txt', encoding="utf-8") as f:
  directions = list(f.read())

floor = 0
for d in directions:
    if d == GO_UP:
        floor += 1
    else:
        floor -= 1

print("Part 1: ", floor)

floor = 0
for idx, d in enumerate(directions):
    if d == GO_UP:
        floor += 1
    else:
        floor -= 1
    if floor == -1:
        print("Part 2: ", idx + 1)
        break

File: 2015/1/puzzle.exs
IO.puts "Day 1: Not Quite Lisp"

case File.read("input.txt") do
  {:ok, contents} ->
    directions = contents
      |> String.split("", trim: true)

    floors = directions
    |> Enum.scan(0, fn d, acc ->
      case d do
        "(" -> acc + 1
        _ -> acc - 1
      end
    end)

    floors
    |> List.last()
    |> IO.inspect(label: "Part 1")

    floors
    |> Enum.find_index(&(&1 == -1))
    |> Kernel.+(1)
    |> IO.inspect(label: "Part 2")

  {:error, :enoent} -> nil
end
